### ISSUE TITLE
Make our GitHub Actions auto-publish depend on tests passing

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/repo2docker#1006]
Right now I believe that if a tag is created and a release made, then our GitHub Action will automatically create a new PyPI release. What do people think about making this step dependent on our test suite passing, as a sanity check before publishing?

I think the way to do it would be to move this action inside the `yml` config for our tests, and make it depend on them. See here for a pattern we could follow:

https://github.com/executablebooks/jupyter-book/blob/master/.github/workflows/tests.yml#L163

**Comment by manics** [comment]
I think there's a range of views amongst our team :smile:
https://github.com/jupyterhub/team-compass/issues/317

**Comment by betatim** [comment]
I don't think we should block publishing to PyPI because our tests sometimes fail for "random reasons" and the workflow for (re)publishing is a bit tedious. I reckon the fact that making a release is a manual step means we can make use of the human's brain (don't publish if you reckon things are broken, unless of course you know that the broken stuff isn't really broken or what not).

**Comment by manics** [comment]
@choldgraf shall we close this?
